backend/app/controllers/projects.js

```diff
@@ -263,9 +263,73 @@ module.exports = {
+                if (parseInt(updateObj.status) === 1) {
+
+                    // Check that the survey is published
+                    const product = yield thunkQuery(
+                        Product.select().from(Product).where(Product.projectId.equals(req.params.id))
+                    );
+
+                    const survey = yield common.getSurveyFromSurveyService(product[0].surveyId, req.headers.authorization);
+
+                    if (survey.body.status !== 'published') {
+                        throw new HttpError(400, 'Survey is not published, project cannot be started');
+                    }
+
+                    // check that project has at least one subject
+                    const projectUOA = yield thunkQuery(
+                        ProductUOA.select().from(ProductUOA).where(ProductUOA.productId.equals(product[0].id))
+                    );
+
+                    if (!_.first(projectUOA)) {
+                        throw new HttpError(400, 'Project contains no subjects, project cannot be started');
+                    }
+
+                    // check that the project has at least one user
+                    const projectUser = yield thunkQuery(
+                        ProjectUser.select(ProjectUser.star()).from(ProjectUser).where(ProjectUser.projectId.equals(req.params.id))
+                    );
+
+                    if (!_.first(projectUser)) {
+                        throw new HttpError(400, 'No user assigned to project, project cannot be started');
+                    }
+
+                    // check that the project has at least one user group assigned
+                    const projectUserGroup = yield thunkQuery(
+                        ProjectUserGroup.select().from(ProjectUserGroup).where(ProjectUserGroup.projectId.equals(req.params.id))
+                    );
+
+                    if (!_.first(projectUserGroup)) {
+                        throw new HttpError(400, 'No user group assigned to project, project cannot be started');
+                    }
+
+                    // Check stages
+                    const stages = yield thunkQuery(
+                        WorkflowSteps
+                            .select()
+                            .from(
+                                WorkflowSteps
+                                    .leftJoin(Workflow)
+                                    .on(WorkflowSteps.workflowId.equals(Workflow.id))
+                            )
+                            .where(Workflow.productId.equals(product[0].id))
+                    );
+
+                    if (!_.first(stages)) {
+                        throw new HttpError(400, 'No stages assigned to project, project cannot be started');
+                    }
+
+                    for (var i=0; i < stages.length; i++) {
+                        if (stages[i].title === null || stages[i].startDate === null || stages[i].endDate === null) {
```

Comment:
`stages[i].title === null` isn't quite working because we're submitting titles as `""` , so it's slipping by this check. It maybe better to add a check for a blank title.